code/datums/manufacturing_requirements.dm

```diff
@@ -1,216 +1,336 @@
+		switch(src.match_type)
+			if (MATCH_ANY)
+				return (M.getMaterialFlags(M) & src.material_flags) > 0
+			if (MATCH_ALL)
+				return (M.getMaterialFlags(M) & src.material_flags) == src.material_flags
+			if (MATCH_EXACT)
+				return (M.getMaterialFlags(M) == src.material_flags)
+
+/datum/manufacturing_requirement/match_flags/metal
+	name = "Metallic"
+	id = "metal_flag"
+	material_flags = MATERIAL_METAL
+
+/datum/manufacturing_requirement/match_flags/wood
+	name = "Wood"
+	id = "wood_flag"
+	material_flags = MATERIAL_WOOD
 
-/datum/manufacturing_requirement/match_property/rubber
+/datum/manufacturing_requirement/match_flags/rubber
 	name = "Rubber"
 	id = "rubber"
 	material_flags = MATERIAL_RUBBER
 
-/datum/manufacturing_requirement/match_property/wood
-	name = "Wood"
-	id = "wood"
-	material_flags = MATERIAL_WOOD
+/datum/manufacturing_requirement/match_flags/organic_or_rubber
+	name = "Organic or Rubber"
+	id = "organic_or_rubber"
+	material_flags = MATERIAL_ORGANIC | MATERIAL_RUBBER
+
+/datum/manufacturing_requirement/match_flags/fabric
+	name = "Fabric"
+	id = "fabric"
+	material_flags = MATERIAL_RUBBER | MATERIAL_ORGANIC | MATERIAL_CLOTH
+
+/datum/manufacturing_requirement/match_flags/crystal
+	name = "Crystal"
+	id = "crystal"
+	material_flags = MATERIAL_CRYSTAL
+
+/datum/manufacturing_requirement/match_flags/energy
+	name = "Energy Source"
+	id = "energy"
+	material_flags = MATERIAL_ENERGY
+
+/datum/manufacturing_requirement/match_flags/insulated
+	name = "Insulative Material"
+	id = "insulative_flags"
+	material_flags = MATERIAL_CLOTH | MATERIAL_RUBBER
+
+#undef MATCH_ANY
+#undef MATCH_ALL
+#undef MATCH_EXACT
+
+ABSTRACT_TYPE(/datum/manufacturing_requirement/match_subtypes)
+/datum/manufacturing_requirement/match_subtypes
+	VAR_PROTECTED/match_typepath //! The parent type to use for istype() checks
+
+	#ifdef CHECK_MORE_RUNTIMES
+	New()
+		. = ..()
+		if (isnull(src.match_typepath) || !ispath(src.match_typepath))
+			CRASH("[src] has invalid match_typepath [src.match_typepath], it must be non-null and a valid path (not an instance!)")
+	#endif
+
+	is_match(var/datum/material/M)
+		. = ..()
+		if (!.) return
+		if (!(istype(M, src.match_typepath))) return FALSE
+
+/datum/manufacturing_requirement/match_subtypes/gemstone
+	name = "Gemstone"
+	id = "gemstone"
+	match_typepath = /datum/material/crystal/gemstone
+
+/// Manufacturing requirements which check several conditions at once.
+ABSTRACT_TYPE(/datum/manufacturing_requirement/mixed)
+/datum/manufacturing_requirement/mixed
+	VAR_PROTECTED/list/datum/manufacturing_requirement/requirements = list() //! A list of requirements which must all be satisfied for this to return TRUE
+
+	/// Resolve the requirement paths to instances in the cache.
+	New()
+		. = ..()
+		var/list/datum/manufacturing_requirement/requirement_instances = list()
+
+		for (var/requirement_id in src.requirements)
+			requirement_instances += getRequirement(requirement_id)
+		src.requirements = requirement_instances
```

Comment:
Don't much like this. Too tired right now to suggest a better alternative, but to make sure I understand: this is turning a list of string IDs into a list of concrete instances and putting them back on the same variable?

Reply:
Yes -- requirements is initially a list of all the manufacturing requirement datum IDs to use, which get converted into instances defined in the cache, to finish setting up the variable.

This avoids having to call getManufacturingRequirement() constantly during runtime, and either way for a concrete instance it has to access some typepath/id/identifier somehow and use that to get a concrete instance.

Reply:
Sure, my issue is mainly in the "list of type strings" getting converted to "list of concrete instances". I would prefer a different var got used, rather than replacing it. This just seems like a maintenance footgun as-is. Feel free to set the list of strings to null if you think the clean-up is important.